bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/CloneUtils.java

```diff
@@ -83,17 +86,29 @@ public static BError createConversionError(Object value, Type targetType, List<S
         if (errors.isEmpty()) {
             return ErrorUtils.createConversionError(value, targetType);
         }
-        return ErrorUtils.createConversionError(value, targetType, getErrorMessage(errors));
+        return ErrorUtils.createConversionError(value, targetType, getErrorMessage(errors, MAX_CONVERSION_ERROR_COUNT));
     }
 
-    private static String getErrorMessage(List<String> errors) {
-        if (errors.size() == MAX_CONVERSION_ERROR_COUNT + 1) {
-            errors.remove(MAX_CONVERSION_ERROR_COUNT);
-            errors.add("...");
-        }
+    static String getErrorMessage(List<String> errors, int maxErrorCount) {
         StringBuilder errorMsg = new StringBuilder();
-        for (String error : errors) {
-            errorMsg.append("\n\t\t").append(error);
+        int totalErrorCount = errors.size();
+        int tabs = 0;
+        for (int i = 0; i < Math.min(totalErrorCount, maxErrorCount); i++) {
```

Comment:
Why we need this comparison?

Reply:
To show only the first 20 (maxErrorCount) error messages.